### PR TITLE
lcms2: 2.13.1 -> 2.14

### PR DESCRIPTION
**Pull request by anthonyroussel**
###### Description of changes

See https://github.com/mm2/Little-CMS/releases/tag/lcms2.14

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by anthonyroussel** [comment]
Retargeted PR to staging (500+ rebuilds expected)

File: pkgs/development/libraries/lcms2/default.nix [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
   LDFLAGS = if stdenv.hostPlatform.isDarwin then "-Wl,-w" else null;
 
+  passthru.updateScript = nix-update-script {
+    attrPath = pname;
+    extraArgs = [ "--version-regex" "lcms([0-9.]+)" ];
+  };
+
   meta = with lib; {
     description = "Color management engine";
     homepage = "http://www.littlecms.com/";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lcms2/default.nix:28]:
```suggestion
    homepage = "https://www.littlecms.com/";
```

File: pkgs/development/libraries/lcms2/default.nix [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+    owner = "mm2";
+    repo = "Little-CMS";
+    rev = "lcms${version}";
+    sha256 = "sha256-E4D25wRUPNvcaCR9Hx72Hfsb3sc7P5yb8EgfExFxZFg=";
+  };
+
   propagatedBuildInputs = [ libtiff libjpeg zlib ];
 
   # See https://trac.macports.org/ticket/60656
   LDFLAGS = if stdenv.hostPlatform.isDarwin then "-Wl,-w" else null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lcms2/default.nix:19]:
```suggestion
  LDFLAGS = lib.optionalString stdenv.hostPlatform.isDarwin "-Wl,-w";
```

File: pkgs/development/libraries/lcms2/default.nix [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
   outputs = [ "bin" "dev" "out" ];
 
+  src = fetchFromGitHub {
+    owner = "mm2";
+    repo = "Little-CMS";
+    rev = "lcms${version}";
+    sha256 = "sha256-E4D25wRUPNvcaCR9Hx72Hfsb3sc7P5yb8EgfExFxZFg=";
+  };
+
   propagatedBuildInputs = [ libtiff libjpeg zlib ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lcms2/default.nix:16]: Is this still correct?

Comment (anthonyroussel) [reply pkgs/development/libraries/lcms2/default.nix:16]: Looks like yes: https://github.com/mm2/Little-CMS/blob/lcms2.14/configure.ac

File: pkgs/development/libraries/lcms2/default.nix (hunk 2 above)
Comment (anthonyroussel) [reply pkgs/development/libraries/lcms2/default.nix]: Fixed, thanks for the review.

File: pkgs/development/libraries/lcms2/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/lcms2/default.nix:16]: Sorry, I am not that fluent in autoconf. I was referring to the propagation part.

I don't think the pc file requires those explicitly https://github.com/mm2/Little-CMS/blob/master/lcms2.pc.in

**Comment by anthonyroussel** [comment]
Superseeded by https://github.com/NixOS/nixpkgs/pull/220271

**Pull request closed without merging**